### PR TITLE
test(lh-66228): add CI pipeline for terraform

### PR DESCRIPTION
**Pull request by siddhuwarrier**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -9,15 +17,11 @@ data "aws_ami" "sec" {
     values = ["x86_64"]
   }
 
-  owners      = ["692314432491"]
+  owners = ["692314432491"]
 
   most_recent = true
 }
 
-data "aws_vpc" "sec" {
```
Comment (doronlevari) [inline main.tf:17]: Nothing expect this?

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+              with:
+                  tflint_version: ${{ env.TFLINT_VERSION }}
+            - name: Init TFLint
+              run: tflint --init
+            - name: Run TFLint
+              run: tflint -f compact
+            - name: Terraform Format Check
+              id: format
+              run: terraform fmt -check -recursive -no-color
+              continue-on-error: true
```
Comment (doronlevari) [inline .github/workflows/pull-request.yml:43]: Awesome how simple this is!!

**Review by doronlevari — APPROVED** [review]
LGTM

File: main.tf (hunk 1 above)
Comment (siddhuwarrier) [reply main.tf:17]: Nope, tflint spotted it. I was previously using this to restrict the security group on the EC2 instance to traffic just from the VPCs. But NLBs are weird; they don't have their own SGs, they rely on the SGs of the EC2 instances. So if I block external traffic from the EC2 instance, traffic from outside the VPC won't flow in. 

The EC2 instance is however on the private subnet, so even if it is open to the world, it is not possible to reach it unless you go through the NLB.

File: .github/workflows/pull-request.yml (hunk 2 above)
Comment (siddhuwarrier) [reply .github/workflows/pull-request.yml:43]: I love Github Actions

**Pull request merged**